modules/kernel/src/main/scala-2.10/notebook/ReplCalculator.scala

```diff
@@ -193,11 +193,11 @@ class ReplCalculator(
         if (queue.nonEmpty) {
           //queue could be empty if InterruptRequest was asked!
           log.debug("Dequeuing execute request current size: " + queue.size)
-          queue = queue.dequeue._2
-          queue.headOption foreach { case (ref, er) =>
-            log.debug("About to execute request from the queue")
-            execute(ref, er)
-          }
+          val (executeRequest, newQueue) = queue.dequeue
+          queue = newQueue
+          val (ref, er) = executeRequest
```

Comment:
I agree, I was thinking the same. So having a separate variable for the current executing request is probably better.
And again, since we're in an actor there will not be any sync problem.

So we'll can use this variable to cancel the current one, and notify the reference actor. But for the case we intereupt everything, we'll need to go over the queue, which is fair